src/main/java/io/micronaut/gradle/graalvm/MicronautGraalPlugin.java

```diff
@@ -135,6 +161,58 @@ public void apply(Project project) {
                     nativeImageTestTask.setDescription("Runs tests against a native image build of the server. Requires the server to allow the port to configurable with 'micronaut.server.port'.");
             })));
 
+            tasks.withType(AbstractCompile.class, compileTask -> {
+                compileTask.doLast(task -> {
```

Comment:
I needed to hook into this task instead of `ProcessResources` because with Kotlin + OpenAPI the files are generated after the `processResources` task. With Java it works as expected using `ProcessResources` task.

The same also happened in the Maven plugin and I hooked into the `prepare-package` phase.

Reply:
This should probably not be implemented as a task action, and probably not on `AbstractCompile` either. Basically this would imply that compilation would generate resources, which means too much work for consumers which do not need those resources.

Ideally this should be a task instead. Would you mind explaining what this is doing an what you're trying to do?

Reply:
I was looking forward for you to review this :-)

We need to generate GraalVM `resource-config.json` file for declaring all the resources that need to be included in the native image. 
In this same PR I "converted" `MicronautGraalPlugin` to a real plugin so users could apply it manually in case they don't use our Micronaut Application plugin.

The best scenario is that this code is only executed when creating the application fatjar or when running the `nativeImage`, `buildDockerNative`, and related tasks; i.e: only when is actually needed, as you said.

Reply:
@melix let's discuss this tomorrow in a call